front/src/applications/operationalStudies/components/SimulationResults/SpeedSpaceChart/SpeedSpaceChart.tsx

```diff
@@ -57,27 +71,50 @@ export default function SpeedSpaceChart(props) {
   const [rotate, setRotate] = useState(false);
   const [showSettings, setShowSettings] = useState(false);
 
-  const ref = useRef();
+  const ref = useRef<HTMLDivElement>(null);
 
-  const onLocalSetSettings = (settings) => {
+  const dispatch = useDispatch();
+
+  const dispatchUpdateMustRedraw = (newMustRedraw: boolean) => {
+    dispatch(updateMustRedraw(newMustRedraw));
+  };
```

Comment:
Also I know you can remove all dispatchUpdateMustRedraw from this component, as I have removed all the behaviour associated to it :) (I let you test everything works without it, I have already tested on my side and it seems ok)
```suggestion
```

Reply:
Good to know ! It works for me as well :)